src/amt/ConnectedDevice.ts

```diff
@@ -63,6 +63,22 @@ export class ConnectedDevice {
     return pullResponse
   }
 
+  async getSoftwareIdentity (): Promise<any> {
+    let xmlRequestBody = this.cim.SoftwareIdentity(CIM_Methods.ENUMERATE, (this.messageId++).toString())
+    const result = await this.ciraHandler.Enumerate(this.ciraSocket, xmlRequestBody)
+    const enumContext: string = result?.Envelope.Body?.EnumerateResponse?.EnumerationContext
+    if (enumContext == null) {
+      logger.error('failed to pull CIM_SoftwareIdentity in get version')
+      return null
+    }
+    xmlRequestBody = this.cim.SoftwareIdentity(CIM_Methods.PULL, (this.messageId++).toString(), enumContext)
+    const pullResponse = await this.ciraHandler.Pull<CIM_SoftwareIdentity>(this.ciraSocket, xmlRequestBody)
+    if (pullResponse == null) {
+      logger.error('failed to pull CIM_SoftwareIdentity in get version')
+      return null
+    }
+    return pullResponse
```

Comment:
don't we want to update this to unwrap `envelope.body`?

Reply:
Will make an other PR with the same change for all the calls and update unit tests